### PR TITLE
{Compute} `az vm/vmss create`: Add warning log to recommend users to specify the `--enable-secure-boot True` and `--enable-vtpm True` when the `--security-type` used by the VM/VMSS creation is `TrustedLaunch`

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Related command**

`az vm create`, `az vmss create`

**Description**

When the `--security-type` used by the VM/VMSS creation is `TrustedLaunch`, print the warning log to recommend users to specify the `--enable-secure-boot True` and `--enable-vtpm True` first.
image
image
image
Close https://github.com/Azure/azure-cli/issues/23144

**Testing Guide**

**History Notes**

{Compute} `az vm/vmss create`: Add warning log to recommend users to specify the `--enable-secure-boot True` and `--enable-vtpm True` when the `--security-type` used by the VM/VMSS creation is `TrustedLaunch`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 1]
```diff
@@ -1311,6 +1311,16 @@ def _enable_msi_for_trusted_launch(namespace):
             namespace.assign_identity.append(MSI_LOCAL_ID)
 
 
+def _validate_trusted_launch(namespace):
+    if not namespace.security_type:
+        return
+
+    if namespace.security_type.lower() == 'trustedlaunch' and \
+            (namespace.enable_vtpm is not True or namespace.enable_secure_boot is not True):
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:1319]:
```suggestion
    if not namespace.security_type or namespace.security_type.lower() != 'trustedlaunch':
        return

    if not namespace.enable_vtpm or not namespace.enable_secure_boot:
```

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 2]
```diff
@@ -1311,6 +1311,16 @@ def _enable_msi_for_trusted_launch(namespace):
...
+        logger.warning('Please set --enable-secure-boot to True and --enable-vtpm to True in order to receive the full'
+                       ' suite of security features that comes with Trusted Launch.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:1321]:
```suggestion
        logger.warning('It is recommended to specify "--enable-secure-boot True" and "--enable-secure-boot True" to receive the full suite of security features that comes with Trusted Launch.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 3]
```diff
@@ -1675,6 +1686,7 @@ def process_vmss_create_namespace(cmd, namespace):
     _validate_vmss_create_nsg(cmd, namespace)
     _validate_vm_vmss_accelerated_networking(cmd.cli_ctx, namespace)
     _validate_vm_vmss_create_auth(namespace, cmd)
+    _validate_trusted_launch(namespace)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:1689]: @sirfamelin May I ask do we need to add this warning log for flex VMSS?

Comment (sirfamelin) [reply src/azure-cli/azure/cli/command_modules/vm/_validators.py:1689]: Yes

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**